### DISCUSSION TITLE
Anything different when using Sail?

### DISCUSSION THREAD

**Discussion by axtg** [larastan/larastan#1286] [Q&A] [answered]
So I started using Larastan today. Ran it, and am getting a lot of `Access to an undefined property App\Http\Resources\SomeResource::$id`. 

But, makes sense! 

So I add `/** @extends JsonResource<\App\Models\SomeModel> */` and run it again. Same thing. I went through some tickets here and found that there are two requirements for Larastan to identify properties automatically: 

1. Have Models extend `use Illuminate\Database\Eloquent\Model;`
2. Have migrations in `database/migrations`

I stick to both conventions. Yet, still my properties are not recognized. Now I know I can add the ide-helper and such, but I'd prefer for Larastan to work as intended.

The only thing I can think of is that Sail might impact my setup? Yet both `composer phpstan` and `sail composer phpstan` do not work. So what else could it be? Any suggestions?

**Comment by canvural** [comment, marked as answer]
Hi,

`JsonResource` is not generic anymore. I don't know what version of Larastan you are using. But if you are using the latest version 2, you can add `/** @mixin \App\Models\SomeModel */` to your API resources to get rid of those errors.

https://github.com/nunomaduro/larastan/blob/37f9ce80890a67a02b1d55bf79eb73604880c6b9/tests/Application/app/UserResource.php#L7-L8
